nova/virt/vmwareapi/vmops.py

```diff
@@ -965,6 +974,16 @@ def _fetch_image_from_other_datastores(self, vi):
             tmp_vi.datastore = ds
             other_templ_vm_ref = self._find_image_template_vm(tmp_vi)
             if other_templ_vm_ref:
+                vmdk = vm_util.get_vmdk_info(self._session, other_templ_vm_ref)
+                if vmdk.capacity_in_bytes == images.INVALID_VMDK_SIZE:
+                    LOG.warning("Deleting invalid template VM %s",
+                                vutil.get_moref_value(other_templ_vm_ref))
+                    destroy_task = self._session._call_method(
+                        self._session.vim,
+                        "Destroy_Task",
+                        other_templ_vm_ref)
+                    self._session._wait_for_task(destroy_task)
+                    return None
```

Comment:
I think we should rather `continue` instead of `return None` here to try and fetch a template from one of the other locations we might have in `dc_other_datastores`